10up/windows-azure-storage
# Add PDF replacement feature

### Description of the Change
We've seen some situations where users want to replace a PDF file with an updated version (CV for example)
This PR adds to the plugin the ability to replace pdf files (for now) using a new method to copy the blob and replace it with the uploaded file.
There's a new method called `copy_media_to_blob_storage`  in helpers and also a new `copy_blob` method in the API client
this uses the Copy Blob operation described here: https://learn.microsoft.com/en-us/rest/api/storageservices/Copy-Blob

There's also a new button added in the media uploader that holds the workflow to replace the document


## Open discussion for addressing images
We should have a discussion on how to approach image replacements since an image of a different orientation can be uploaded and image sizes of the image to be replaced won't match, so I was thinking on two paths
-  Should we force the same image sizes from the replaced image so we can have backward compatibility in case those images are being called directly?
- Should we discard previous images and just replace thumbnails with the ones generated by the new image?


### How to test the Change
To test this:
- Go to the media screen in the admin dashboard (`wp-admin/upload.php`)
- Upload a PDF file
- Select the file and you'll see a "replace this media" button
- Click the button and select the file to upload (the view is restricted to just upload a file)
- Once it's uploaded click on the "Replace media" button below
- That will trigger the replacement of the main file and the thumbnails stored as meta fields

### Changelog Entry


> Added - New feature to replace PDF files at the blob storage level
> Fixed - Bug fix. Fixed a bug with the transient generated for displaying progress



### Credits
@hugosolar 


### Checklist:
